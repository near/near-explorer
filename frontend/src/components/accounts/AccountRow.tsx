import moment from "moment";
import Link from "next/link";

import React from "react";
import { Row, Col } from "react-bootstrap";

import AccountsApi from "../../libraries/explorer-wamp/accounts";

import Balance from "../utils/Balance";

export interface Props {
  accountId: string;
  createdAt: number;
}

export interface State {
  amount: string;
}

export default class extends React.Component<Props, State> {
  state: State = {
    amount: ""
  };

  _getDetail = async () => {
    const detail = await new AccountsApi().queryAccount(this.props.accountId);
    this.setState({ amount: detail.amount });
  };

  componentDidMount() {
    this._getDetail();
  }

  render() {
    const { accountId, createdAt } = this.props;
    const { amount } = this.state;
    return (
      <Link href="/accounts/[id]" as={`/accounts/${accountId}`}>
        <a style={{ textDecoration: "none" }}>
          <Row className="transaction-row mx-0">
            <Col md="auto" xs="1" className="pr-0">
              <img
                src="/static/images/icon-t-acct.svg"
                style={{ width: "15px" }}
              />
            </Col>
            <Col md="7" xs="6">
              <Row>
                <Col className="transaction-row-title">@{accountId}</Col>
              </Row>
            </Col>
            <Col md="3" xs="4" className="ml-auto text-right">
              <Row>
                <Col className="transaction-row-txid">
                  {amount && <Balance amount={amount} />}
                </Col>
              </Row>
              <Row>
                <Col className="transaction-row-timer">
                  {createdAt && `Created at ${moment(createdAt).format("LLL")}`}
                </Col>
              </Row>
            </Col>
            <style jsx global>{`
              .transaction-row {
                padding-top: 10px;
                padding-bottom: 10px;
                border-top: solid 2px #f8f8f8;
              }

              .transaction-row:hover {
                background: rgba(0, 0, 0, 0.1);
              }

              .transaction-row-title {
                font-family: BentonSans;
                font-size: 14px;
                font-weight: 500;
                line-height: 1.29;
                color: #24272a;
              }

              .transaction-row-text {
                font-family: BentonSans;
                font-size: 12px;
                font-weight: 500;
                line-height: 1.5;
                color: #999999;
              }

              .transaction-row-txid {
                font-family: BentonSans;
                font-size: 14px;
                font-weight: 500;
                line-height: 1.29;
                color: #0072ce;
              }

              .transaction-row-timer {
                font-family: BentonSans;
                font-size: 12px;
                color: #999999;
                font-weight: 100;
              }

              .transaction-row-timer-status {
                font-weight: 500;
              }
            `}</style>
          </Row>
        </a>
      </Link>
    );
  }
}
