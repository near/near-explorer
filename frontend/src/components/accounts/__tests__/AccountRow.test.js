import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });
import BN from "bn.js";

import AccountRow from "../AccountRow";

describe("<AccountRow />", () => {
  beforeEach(() => jest.resetAllMocks());

  it("should load amount", async () => {
    const wrapper = shallow(
      <AccountRow
        accountId="account"
        createdAt={Number(new Date(2019, 1, 1))}
      />
    );

    const amount = new BN("100000000000000000000000000").add(
      new BN("1000000000000000000000000")
    );

    wrapper.instance().state.amount = amount.toString();
    wrapper.update();

    expect(wrapper.instance().state.amount).toEqual(
      "101000000000000000000000000"
    );
  });

  it("renders", () => {
    expect(
      renderer.create(
        <AccountRow
          accountId="account"
          createdAt={Number(new Date(2019, 1, 1))}
        />
      )
    ).toMatchSnapshot();
  });
});
