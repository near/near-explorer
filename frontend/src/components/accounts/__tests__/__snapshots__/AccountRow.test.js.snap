// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountRow /> renders 1`] = `
<a
  href="/accounts/account"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <div
    className="transaction-row mx-0 row"
  >
    <div
      className="pr-0 col-md-auto col-1"
    >
      <img
        className="jsx-324075529"
        src="/static/images/icon-t-acct.svg"
        style={
          Object {
            "width": "15px",
          }
        }
      />
    </div>
    <div
      className="col-md-7 col-6"
    >
      <div
        className="row"
      >
        <div
          className="transaction-row-title col"
        >
          @
          account
        </div>
      </div>
    </div>
    <div
      className="ml-auto text-right col-md-3 col-4"
    >
      <div
        className="row"
      >
        <div
          className="transaction-row-txid col"
        >
          
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="transaction-row-timer col"
        >
          Created at February 1, 2019 12:00 AM
        </div>
      </div>
    </div>
  </div>
</a>
`;
